### DISCUSSION TITLE
localization only works for 2 letter locales?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#5154] [Help] [answered]
(I tried submitting a bug, but the bot closed it because I don't have a repository)

#5153 

My `app.cfg` has set the locale to Dutch
```
'locale' => 'nl_NL'
```

This works fine in Laravel / Carbon.
I need the full locale (`nl_NL`) because that one uses the correct short day names (and the 2 letter locale `nl` adds unwanted periods to the day names).

Issue: the Filament interface ("per page", "next" etc.) and e.g. DatePicker use the fallback_locale: in my case `en`.
When I set locale to the 2 letter locale `nl` everything works fine.

Same for e.g. German: `de_DE` and `de`.

Is there a reason why Filament does not accept the full locale name?

**Comment by danharrin** [comment, marked as answer]
Because Laravel does not fallback `nl_NL` to `nl` if it is not present in our language files. We only have a `lang/nl` directory, and copying files to `lang/nl_NL` also would be unmaintainable. Ultimately this is an issue with Carbon and there is little we or Laravel can do about it.

**Reply by wivaku** [reply]
Thanks @danharrin! For the Filament specific translations, understood: there are no nl_NL translations.
For the DatePicker: would it not use Carbon and therefore be able to show the day/week names correctly?

**Reply by danharrin** [reply]
DatePicker uses Day.js for formatting, not Carbon, because it's powered by JS

**Reply by wivaku** [reply]
Of course, that makes sense. Thanks again Dan.
